subt/dummy_darpa_server.py

```diff
@@ -0,0 +1,92 @@
+#!/usr/bin/env python
+"""
+  webserver with connection to serial controller
+  Usage:
+     ./web2.py [<COM port>|dummy]
+"""
+
+#import string,cgi,time
+import os
+import sys
+import csv
+import math
+import json
+from collections import defaultdict
+from http.server import BaseHTTPRequestHandler, HTTPServer
+from mimetypes import guess_type
+
+
+def dist3d(xyz, xyz2):
+    return math.sqrt(sum([(a-b)**2 for a, b in zip(xyz, xyz2)]))
+
+
+class GameLogic:
+    def __init__(self, filename):
+        self.score = 0
+        self.artf = defaultdict(list)
+        with open(filename) as csvfile:
+            reader = csv.reader(csvfile)
+            for raw in reader:
+                if 'artf' in raw:
+                    continue
+                self.artf[raw[0]].append(tuple([float(x) for x in raw[1:]]))
```

Comment:
This line is not very readable. Additionally, the `tuple` call can take the generator directly, no need to create a list using `[]` first.